### PR TITLE
ci: set up OIDC in the release workflow

### PR DESCRIPTION
**Pull request by reakaleek**
This sets up keyless authentication to GCP with OIDC.

I have a successful dry-run here: https://github.com/elastic/apm-agent-rum-js/actions/runs/10850931893, which is deploying to a test bucket in the elastic-observability-ci GCP project

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -15,8 +15,9 @@ permissions:
   contents: read
 
 env:
-  ELASTIC_CDN_BUCKET_NAME: ${{ inputs.dry-run == false && 'apm-rum-357700bc' || 'oblt-apm-agent-rum-js-ci' }}
-  ELASTIC_CDN_CREDENTIALS: ${{ inputs.dry-run == false && 'secret/gce/elastic-cdn/service-account/apm-rum-admin' || 'secret/observability-team/ci/service-account/apm-agent-rum-js' }}
+  ELASTIC_CDN_BUCKET_NAME: ${{ inputs.dry-run == false && 'apm-rum-357700bc' || 'oblt-apm-rum-test' }}
+  ELASTIC_CDN_PROJECT_ID: ${{ inputs.dry-run == false && 'elastic-cdn-4ae000ab' || 'elastic-observability-ci' }}
+  ELASTIC_CDN_PROJECT_NUMBER: ${{ inputs.dry-run == false && '382950469386' || '911195782929' }}
```
Comment (v1v) [inline .github/workflows/release.yml:20]: There is no way to use one or the other but both?

**Review by v1v — DISMISSED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (reakaleek) [reply .github/workflows/release.yml:20]: Thanks for the hint.

Seems like in existing cases we are only using the project-number.

Comment (reakaleek) [reply .github/workflows/release.yml]: https://github.com/elastic/apm-agent-rum-js/pull/1540/commits/4f8a3f569e1baeab5eea3cb164bc83032c8bbbf4

The dry-run of that commit was successful here: https://github.com/elastic/apm-agent-rum-js/actions/runs/10879975420

**Review by v1v — DISMISSED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -61,21 +62,11 @@ jobs:
           echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > .npmrc
           npm run ci:release
 
-      - name: Read GCE vault secrets
-        uses: hashicorp/vault-action@v3.0.0
+        # The permissions configuration for this can be found at
+        # https://github.com/elastic/oblt-infra/blob/main/conf/resources/repos/apm-agent-rum-js/02-gcp-oidc.tf
```
Comment (reakaleek) [inline .github/workflows/release.yml:66]:
```suggestion
        # https://github.com/elastic/oblt-infra/blob/main/conf/resources/repos/apm-agent-rum-js/02-gcp-oidc-elastic-cdn.tf
        # and
        # https://github.com/elastic/oblt-infra/blob/main/conf/resources/repos/apm-agent-rum-js/02-gcp-oidc-elastic-observability.tf
```

**Review by v1v — APPROVED** [review]

**Review by vigneshshanmugam — APPROVED** [review]

**Pull request merged**